pandas/core/arrays/datetimes.py

```diff
@@ -603,13 +610,18 @@ def __iter__(self):
             length = len(self)
             chunksize = 10000
             chunks = (length // chunksize) + 1
-            for i in range(chunks):
-                start_i = i * chunksize
-                end_i = min((i + 1) * chunksize, length)
-                converted = ints_to_pydatetime(
-                    data[start_i:end_i], tz=self.tz, freq=self.freq, box="timestamp"
-                )
-                yield from converted
+
+            with warnings.catch_warnings():
+                # filter out warnings about Timestamp.freq
+                warnings.filterwarnings("ignore", category=FutureWarning)
+
+                for i in range(chunks):
+                    start_i = i * chunksize
+                    end_i = min((i + 1) * chunksize, length)
+                    converted = ints_to_pydatetime(
+                        data[start_i:end_i], tz=self.tz, freq=self.freq, box="timestamp"
```

Comment:
can you just NOT pass self.freq?

Reply:
Not without changing the behavior.  ATM `next(iter(dti)).freq` matches `dti.freq`.  Not passing `freq` here would change that